cmd/estclient/cacerts.go

```
@@ -52,18 +52,26 @@ func cacerts(w io.Writer, set *flag.FlagSet) error {
 		return fmt.Errorf("failed to get CA certificates: %v", err)
 	}
 
-	if cfg.FlagWasPassed(rootOutFlag) {
-		var root *x509.Certificate
+	// filter certificates if requested
+	if cfg.FlagWasPassed(rootsOnlyFlag) != cfg.FlagWasPassed(intermediatesOnlyFlag) {
+		var roots, intermediates []*x509.Certificate
+
 		for _, cert := range certs {
-			if bytes.Equal(cert.RawSubject, cert.RawIssuer) && cert.CheckSignatureFrom(cert) == nil {
-				root = cert
-				break
+			// self-issued by RFC 5280 (same subject and issuer; that includes self-signed)
+			if bytes.Equal(cert.RawSubject, cert.RawIssuer) {
+				roots = append(roots, cert)
+			} else {
+				intermediates = append(intermediates, cert)
 			}
 		}
-		if root == nil {
-			return errors.New("failed to find a root certificate in CA certificates")
+
+		if cfg.FlagWasPassed(rootsOnlyFlag) {
+			certs = roots
+		} else if cfg.FlagWasPassed(intermediatesOnlyFlag) {
+			certs = intermediates
 		}
-		certs = []*x509.Certificate{root}
+	} else if cfg.FlagWasPassed(rootsOnlyFlag) && cfg.FlagWasPassed(intermediatesOnlyFlag) {
+		return errors.New("Specifying both '-intermediates' and '-roots' doesn't make sense - use one of them or none")
```

This error message comes across as a bit harsh - something like:

    return fmt.Errorf("only one of -%s and -%s may be specified", rootsOnlyFlag, intermediatesOnlyFlag)

would be better.

Not so much of a big deal, but it would probably also be better to move this check earlier in the function, so we can fail early if the options conflict.

---

Yeah, I agree its a bit harsh 😄 
Does it make sense to move it to `newConfig()` maybe?

---

moved to the beginning of the function and rephrased as suggested